### ISSUE TITLE
Core dumped after checking failed

### ISSUE THREAD

**Issue by candou1611** [tensorflow/tensorflow#16761]
### System information
== cat /etc/issue ===============================================
Linux 84a2861a8739 3.10.0-229.el7.x86_64 #1 SMP Fri Mar 6 11:36:42 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
VERSION="7 (Core)"
VERSION_ID="7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

== are we in docker =============================================
Yes

== compiler =====================================================
c++ (GCC) 4.8.5 20150623 (Red Hat 4.8.5-4)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

== uname -a =====================================================
Linux 84a2861a8739 3.10.0-229.el7.x86_64 #1 SMP Fri Mar 6 11:36:42 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

== check pips ===================================================
numpy (1.13.3)
protobuf (3.5.0.post1)
tensorflow-tensorboard (0.4.0rc3)

== check for virtualenv =========================================
False

== tensorflow import ============================================
tf.VERSION = 1.5.0
tf.GIT_VERSION = v1.5.0-0-g37aa430d84
tf.COMPILER_VERSION = v1.5.0-0-g37aa430d84
Sanity check: array([1], dtype=int32)

== env ==========================================================
LD_LIBRARY_PATH is unset
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
tf_env_collect.sh: line 105: nvidia-smi: command not found

== cuda libs  ===================================================

### Describe the problem
I used keras using TensorFlow backend to train a batch of 300\*300\*3 RGB image.
The program occurred core dumped after printing check failed log as below.

### Source code / logs
F tensorflow/core/kernels/maxpooling_op.cc:177] Check failed: input_backprop_index >= in_start && input_backprop_index < in_end Invalid input backprop index: -1491167680, 2803712000, 2806515712

I traced tensorflow source code, it should be check operation. And backprop index shouldn't be negative. I don't know tensorflow well, why can appear such problem? I think this error is root cause of code dumped. Could you help me solve this problem?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by candou1611** [comment]
Thanks for your update. Here are fields you need.
Have I written custom code. N/A
OS Platform and Distribution. Centos7 and Linux 84a2861a8739 3.10.0-229.el7.x86_64
TensorFlow installed from. Using PIP
TensorFlow version. 1.4/1.5
Bazel version. N/A
CUDA/cuDNN version. N/A
GPU model and memory. N/A
Exact command to reproduce.  Using Keras (Tensorflow as backend)

**Comment by skye** [comment]
Can you post a script we can run to reproduce this problem?

cc/ @vrv , I'm not sure who else should handle this

**Comment by vrv** [comment]
@skye I think the problem is that the input_backprop_index is an int, not an int64.

**Comment by candou1611** [comment]
@skye @vrv , Thanks for your comments.
I already attach script in [here](https://stackoverflow.com/questions/48574868/tensorflow-core-dumped-after-check-failed), while a batch of training image is very large, it's not possible to upload them.
I just change input_backprop_index from int to int64 in 172 line located in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/maxpooling_op.cc
Right?

**Comment by bignamehyp** [comment]
Has this issue been addressed?

**Comment by rpp0** [comment]
I'm having a similar issue. It seems like there is an integer overflow in `maxpooling_op.cc` when training on large images. I'm currently mitigating this by reducing the resolution of the input images.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by felonwan** [comment]
> @skye @vrv , Thanks for your comments.
> I already attach script in [here](https://stackoverflow.com/questions/48574868/tensorflow-core-dumped-after-check-failed), while a batch of training image is very large, it's not possible to upload them.
> I just change input_backprop_index from int to int64 in 172 line located in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/maxpooling_op.cc
> Right?

Have you solved this issue?

I have a similar error. I changed int to int64 from line 165 to line 172 in tensorflow/blob/master/tensorflow/core/kernels/maxpooling_op.cc, then compiled tensorflow using bazel and install the whl file. But it still does not work.
